src/TestExplorer/TestRunner.ts

```diff
@@ -193,8 +190,15 @@ export class TestRunProxy {
         this.testRun?.errored(test, message, duration);
     }
 
-    public end() {
-        this.testRun?.end();
+    public async end() {
+        if (!this.testRun) {
+            return;
+        }
+
+        // Compute final coverage numbers if any coverage info has been captured during the run.
+        await this.coverage.computeCoverage(this.testRun);
```

Comment:
How about moving this to where the coverage `createRunProfile` is created instead of calling for every test run